### PR TITLE
#3178 Add more failure info to QueryCompletionEvent

### PR DESCRIPTION
**Pull request by jiachuanwang**
please let me know if you have any question about this commit.

### CODE REVIEW

**Comment by jiachuanwang** [comment]
@martint 
do you think name the failedTask to be failureTask will be a little bit better since all other failure information start with failure.

File: presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java [hunk 1]
```diff
@@ -162,6 +170,12 @@ private void logQueryTimeline(QueryInfo queryInfo)
             long firstTaskStartTime = queryEndTime.getMillis();
             long lastTaskStartTime = queryStartTime.getMillis() + planning.toMillis();
             long lastTaskEndTime = queryStartTime.getMillis() + planning.toMillis();
+
+            TaskInfo task = queryInfo.getOutputStage().getTasks().stream().filter(taskInfo -> taskInfo.getState() == TaskState.FAILED)
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java]: Wrap `.filter()`

File: presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java [hunk 2]
```diff
@@ -128,7 +134,9 @@ public void completionEvent(QueryInfo queryInfo)
                             objectMapper.writeValueAsString(queryInfo.getOutputStage()),
                             objectMapper.writeValueAsString(queryInfo.getFailureInfo()),
                             objectMapper.writeValueAsString(queryInfo.getInputs()),
-                            objectMapper.writeValueAsString(mergedProperties.build())
+                            objectMapper.writeValueAsString(mergedProperties.build()),
+                            taskId,
+                            hostName
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java]: Move the last `));` to the end of this line (it shouldn't have been wrapped)

**Comment by jiachuanwang** [comment]
I accidentally created two more unnecessary commits. please let me know whether need to revert these.

**Comment by jiachuanwang** [comment]
Hi @electrum, do you think the code is ok right now? BTW how could I get right to merge the pull request? Thanks

**Comment by jiachuanwang** [comment]
I will make another commit since this change may cause null pointer exception.

**Comment by jiachuanwang** [comment]
Hi @electrum, please let me know if you have any comment for this. Thanks

File: presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java [hunk 3]
```diff
@@ -96,6 +97,16 @@ public void completionEvent(QueryInfo queryInfo)
                 }
             }
 
+            TaskInfo task = null;
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java]: This code seems duplicated. How about creating a method `getFailedTask(queryInfo)`.

File: presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java [hunk 4]
```diff
@@ -196,7 +220,7 @@ private void logQueryTimeline(QueryInfo queryInfo)
 
             Duration finishing = millis(queryEndTime.getMillis() - lastTaskEndTime);
 
-            log.info("TIMELINE: Query %s :: elapsed %s :: planning %s :: scheduling %s :: running %s :: finishing %s :: begin %s :: end %s",
+            log.info("TIMELINE: Query %s :: elapsed %s :: planning %s :: scheduling %s :: running %s :: finishing %s :: begin %s :: end %s :: failure task %s :: failure host %s",
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java]: I don't think we need to add this to the timeline.

**Comment by electrum** [comment]
Overall looks good. Please address comments and squash the commits. Then we will merge it.

**Pull request closed without merging**